src/cljs/velho_ds/molecules/field.cljs

```diff
@@ -145,127 +167,127 @@
 
-
 (defn dropdown-menu [{:keys [label
                              placeholder
                              item-list
-                             on-change-fn
-                             on-blur-fn
-                             on-focus-fn
-                             on-item-select-fn
-                             selected-item
-                             icon-click-fn
-                             icon
-                             error-messages
+                             selected-fn
+                             preselected-value
                              disabled
                              styles]}]
-  (let [item-list-keys (atom [])
-        state (r/atom {:id (subs (str (rand)) 2 9)
-                       :input-text (:label selected-item)
-                       :selected-item (if selected-item [selected-item] [])
-                       :selected-idx nil
-                       :items []
-                       :item-count (apply + (map #(count (:items %)) item-list))
-                       :focus false
-                       :disabled (if disabled disabled false)})
+  (assert (fn? selected-fn) ":selected-fn function is required for dropdown-menu")
+  (assert (vector? item-list) ":item-list vector is required for dropdown-menu")
+  (let [dropdown-id (atom (str (random-uuid)))
 
-        change (fn [val]
-                 (when (not (nil? on-change-fn))
-                   (swap! state assoc :selected-idx nil)
-                   (swap! state assoc :input-text val)
-                   (when on-change-fn (on-change-fn (:input-text @state)))))
+        state (r/atom (let [itemlist (into [{:items [{:label (if placeholder placeholder "")}]}] item-list)
+                            flatten-itemlist (flatten (map (fn [item] (get item :items)) (filter-items itemlist "")))]
+                        {:items itemlist
+                         :input-text ""
+                         :items-filtered flatten-itemlist
+                         :selected-item (if preselected-value preselected-value nil)
+                         :selected-idx (if preselected-value (index-of-item preselected-value flatten-itemlist) 0)
+                         :selected-from-filter (if preselected-value preselected-value placeholder)
+                         :focus false
+                         :disabled (if disabled disabled false)}))
 
-        blur (fn []
-               (swap! state assoc :focus false)
-               (if (some #(some (fn [a] (= (:input-text @state) (:label a))) (:items %)) item-list)
-                 (swap! state assoc :selected-item (:input-text @state))
-                 (swap! state assoc :input-text (:selected-item @state)))
-               (when on-blur-fn (on-blur-fn (:input-text @state)))
-               (.blur (dommy/sel1 (keyword (str "#input-dropdown-menu-" (:id @state))))))
+        style-list-item (fn [item]
+                          (when (= (:label item) placeholder)
```

Comment:
Placeholder shouldn't be confused with actual value. This could cause problem if list contains item that has same value as placeholder